### PR TITLE
Runtime & Performance analysis: Week 3&4

### PR DESCRIPTION
**Pull request by shdshdw**

### CODE REVIEW

File: 08-runtime-and-performance-analytics.Rmd [hunk 1]
```diff
@@ -7,15 +7,23 @@ cover the entire field because it is too broad. Using Kitchenham’s method
...
 
-We have gathered a few other papers (excluding the five initial papers) to find out if this field
-is suited for this survey. These papers can be found in Figure INSERT FIGURE NUMBER HERE.
-To get relevant papers, we used the following keywords: Android, performance, runtime, reliability,
-synchronization, security, monitoring. Furthermore, we only retrieved papers published at top
-venues, which we list here:
+## Introduction
+Energy consumption plays a big role in day-to-day usage of software. Especially in the field of software development for mobile devices, for which energy consumption results in a negative effect on the battery duration and hence limiting the usage time of the device. Although the importance of energy efficient software is clear, programmers lack the knowledge of the best practices to reduce software energy consumption, and education is not focused on this field either [ref:PHA2016]. In an analysis of questions posted on StackOverflow on the topic of energy efficiency performed by Pinto et. al. [ref:PCL2014] it was found that although programmers had questions related to energy efficiency, they rarely got the appropriate advice.
```
Comment (qurben) [inline 08-runtime-and-performance-analytics.Rmd]: Please wrap your lines

Comment (qurben) [reply 08-runtime-and-performance-analytics.Rmd]: `[ref:PHA2016]` doesn't work in bookdown, check out https://bookdown.org/yihui/bookdown/citations.html

File: 08-runtime-and-performance-analytics.Rmd [hunk 2]
```diff
@@ -28,161 +36,76 @@ venues, which we list here:
...
-Which programming languages are the most EDP efficient and inefficient for particular tasks?
-Which types of programming languages are, on average, more EDP efficient and inefficient for each of the selected platforms (i.e. server, laptop and embedded system)?
-How much does the EDP of each programming language differ among the selected platforms?
-To answer these questions the Rosetta Code Repository, a publicly available repository for programming tasks, is used. It offers 868 tasks, 204 draft tasks and has implementations in 675 programming languages.
-The results of the paper are that for most tasks the compiled programming languages outperform the interpreted ones.
+Another comparable recent research paper investigated the energy-delay product (EDP) which is defined as  “a weighted function of the energy consumption and run-time performance product” [ref:GKLS2018] :\s\s
```
Comment (qurben) [inline 08-runtime-and-performance-analytics.Rmd]: `\s\s` makes my Latex crash, check out https://bookdown.org/yihui/bookdown/markdown-syntax.html#math-expressions on how to do inline latex.

File: 08-runtime-and-performance-analytics.Rmd [hunk 3]
```diff
@@ -7,15 +7,23 @@ cover the entire field because it is too broad. Using Kitchenham’s method
 For inspiration, we started reading five recent papers on runtime and performance analytics
 published at top conferences. These five were selected because the papers handle the software side
 of performance and runtime analytics which is more in line with the other chapters of this book.
-However, focussing on only software, the field is still very broad. Currently, we are leaning
-towards a focus on performance and runtime analytics literature regarding the Android platform.
-As we still are at the start of this research, we might deviate from this initial focus.
+However, focussing on only software, the field is still very broad. We have chosen to focus on the field of performance vs. Energy consumption. This chioce was made due to it being a very current and thriving domain within Runtime and performance analysis.
```
Comment (MaxVanDeursen) [inline 08-runtime-and-performance-analytics.Rmd]: Refrain from using abbreviations (e.g. `vs` -> `versus`)
Moreover, `chioce`->`choice`

What do you mean with a current domain?

File: 08-runtime-and-performance-analytics.Rmd [hunk 4]
```diff
@@ -7,15 +7,23 @@ cover the entire field because it is too broad. Using Kitchenham’s method
...
+With over 2 billion daily smartphone and tablet users worldwide [ref:NPPPZL2017] it is clear that this lack of knowledge needs to be addressed. To help the programmers close this knowledge gap, hardware-based tools have been introduced. These tools can determine the energy profile of an application with high precision, but it means the costly hardware components need to be acquired[ref:NPPPZL2017]. Software-based solutions solve the problem of requiring expensive hardware, and are easier to use, but also less precise. 
+
+In this chapter the current state of energy efficiency in software development for mobile devices will be analysed. This research will be conducted by following three research questions:
```
Comment (MaxVanDeursen) [inline 08-runtime-and-performance-analytics.Rmd]: `following` seems a bit of an odd word in this context. Might want to go for `answering`?

File: 08-runtime-and-performance-analytics.Rmd [hunk 5]
```diff
@@ -7,15 +7,23 @@ cover the entire field because it is too broad. Using Kitchenham’s method
...
+
+* **RQ1:** What is the state of the art of energy efficiency in software development for mobile devices?
+* **RQ2:** What is the state of practice for energy efficiency in software development for mobile devices?
+* **RQ3:** What future work needs to be researched in the field of energy efficiency in software development for mobile devices?
+
+In the following paragraph the methodology of the research will be elaborated. After that each research questions will be answered in a separate paragraph. In the last paragraph of this chapter the conclusion on energy efficiency in software development for mobile devices will be made.
+
+
+##Methodology
+In order to find answers to the research questions, over 30 papers in the field have been selected. These papers have been found by searching on Google Scholar with a set of filters. First off, only papers from the following journals and conference are selected:
```
Comment (MaxVanDeursen) [inline 08-runtime-and-performance-analytics.Rmd]: Might be good to explain why only these conferences are chosen?

File: 08-runtime-and-performance-analytics.Rmd [hunk 6]
```diff
@@ -28,161 +36,76 @@ venues, which we list here:
...
-errors.
+The following criteria is given by the activity and relative newness of the field. The preference is given to papers which are published not too long ago, preferably after 2012. But if well cited, and with interesting findings, older papers can also be used.
+
+Given these two criteria the following search queries have been used:
+
+* "Android AND Energy Efficiency AND Software"
+* "Energy AND Android AND Runtime AND Analysis AND performance AND Software"
 
-**Remarks**
+From the selected papers some backward and forward references have also been selected, to get a complete picture. All papers are also checked for relevance to the topic.
```
Comment (MaxVanDeursen) [inline 08-runtime-and-performance-analytics.Rmd]: How do you do this `checking for relevance`?

File: 08-runtime-and-performance-analytics.Rmd [hunk 7]
```diff
@@ -28,161 +36,76 @@ venues, which we list here:
...
 
-- Stefanos Georgiou, Maria Kechagia, Panos Louridas, and Diomidis Spinellis. 2018. What are your programming language's energy-delay implications?. In Proceedings of the 15th International Conference on Mining Software Repositories (MSR '18). ACM, New York, NY, USA, 303-313. DOI: https://doi.org/10.1145/3196398.3196414
+##State of Practice
+Now that we have an overview of the state of the art regarding energy consumption on Android driven mobile devices, we investigate what the current state of practice is. In this section, we try to find an answer to RQ2 stated in the introduction by going through the papers we have retrieved using the methodology described in Section 2. First, we study how different software engineering approaches affect energy consumption and if developers have any knowledge or training regarding energy saving programming techniques. Furthermore, we try to see if there are any tools being used which help decreasing energy consumption. To make this more specific, we divide RQ2 into several sub research questions:
```
Comment (MaxVanDeursen) [inline 08-runtime-and-performance-analytics.Rmd]: For consistency, either always make `RQ*` bold or never do so. Right now, this is done inconsistently.

**Comment by Guillembonet** [comment]
Well organized information and good text

File: 08-runtime-and-performance-analytics.Rmd [hunk 8]
```diff
@@ -28,161 +58,212 @@ venues, which we list here:
...
+implementations in terms of energy consumption and performance. The paper suggests that hybrid 
+implementations might be a good solution, but the problem is that the study was done on just four 
+Android applications. The paper did use the Rosetta Code repository, but this does not represent 
+commercial software development.
+
+Another comparable recent research paper investigated the energy-delay product (EDP) which is 
+defined as  “a weighted function of the energy consumption and run-time performance 
+product” [@GKLS2018]:
+
+$E * T^w$
```
Comment (qurben) [inline 08-runtime-and-performance-analytics.Rmd]: See https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#equations on how to do equations in Bookdown and how to reference them.

**Pull request merged**